script.js

```diff
@@ -458,36 +462,80 @@ function solved() {
 }
 function getDailySeed() {
     const today = new Date();
-    const dateString = today.getFullYear() * 10000 +
-        (today.getMonth() + 1) * 100 +
-        today.getDate();
-
-    const storedSeedData = localStorage.getItem('dailySeed');
-
+    let dateString = today.getFullYear() + today.getMonth() + today.getDate();
+    let storedSeedData = localStorage.getItem('dailySeed');
     if (storedSeedData) {
-        const parsedData = JSON.parse(storedSeedData);
+        let parsedData = JSON.parse(storedSeedData);
+        console.log(parsedData.date);
+        console.log(dateString);
+
 
         if (parsedData.date === dateString) {
-            return parsedData.seed;
+            if (parsedData.isSolved) {
+                return;
+            }
+        }else{
+            const newSeed = Math.floor(Math.random() * 1000000);
+            storedSeedData = {
+                date: dateString,
+                seed: newSeed,
+                isSolved: false
+            };
+            localStorage.setItem('dailySeed', JSON.stringify(storedSeedData));
         }
+    } else {
+        const newSeed = Math.floor(Math.random() * 1000000);
+        storedSeedData = {
+            date: dateString,
+            seed: newSeed,
+            isSolved: false
+        };
+        localStorage.setItem('dailySeed', JSON.stringify(storedSeedData));
     }
+}
 
-    const newSeed = Math.floor(Math.random() * 1000000);
 
-    localStorage.setItem('dailySeed', JSON.stringify({
-        date: dateString,
-        seed: newSeed
-    }));
+function markDailyChallengeAsSolved() {
+    let storedSeedData = localStorage.getItem('dailySeed');
+    if (storedSeedData) {
+        let parsedData = JSON.parse(storedSeedData);
+        parsedData.isSolved = true;
+        localStorage.setItem('dailySeed', JSON.stringify(parsedData));
+    }
+}
+
+
+document.addEventListener('DOMContentLoaded',getDailySeed);
+
+function isdailyChallengeSolved(){
+    const urlParams = new URLSearchParams(window.location.search);
+    const seed = urlParams.get("seed");
+    console.log(seed);
+    const storedSeedData = localStorage.getItem('dailySeed');
+    const storedSeed = JSON.parse(storedSeedData);
+    console.log(storedSeed.seed);
+
+    if (String(storedSeed.seed) === String(seed)) {
+        return true;
+    }
+    else{
+        return false;
+    }
 
-    return newSeed;
 }
 
-function dailychallenge() {
-    const dailySeed = getDailySeed();
-    console.log("mwe");
-    const dailyChallengeLink = document.getElementById('daily-challange');
-        dailyChallengeLink.href = `index.html?seed=${dailySeed}&size=3`;
+const link = document.getElementById("daily-challange");
+link.addEventListener("click", keineAhnungHabKeinNamen)
 
+function keineAhnungHabKeinNamen() {
```

Comment:
Könnte z.B. sowas ein sinnvoller Funktionsname sein?
```suggestion
function onClickDailyChallenge() {
```